internal/core/src/exec/expression/BinaryArithOpEvalRangeExpr.cpp

```diff
@@ -52,24 +52,56 @@ PhyBinaryArithOpEvalRangeExpr::Eval(EvalCtx& context, VectorPtr& result) {
         }
         case DataType::JSON: {
             auto value_type = expr_->value_.val_case();
-            switch (value_type) {
-                case proto::plan::GenericValue::ValCase::kBoolVal: {
-                    result = ExecRangeVisitorImplForJson<bool>();
-                    break;
-                }
-                case proto::plan::GenericValue::ValCase::kInt64Val: {
-                    result = ExecRangeVisitorImplForJson<int64_t>();
-                    break;
-                }
-                case proto::plan::GenericValue::ValCase::kFloatVal: {
-                    result = ExecRangeVisitorImplForJson<double>();
-                    break;
+            if (segment_->type() == SegmentType::Growing) {
```

Comment:
for growing segment, can we also use JsonView and string view to avoid the if else?